### ISSUE TITLE
Uninstalled Event Source results in not properly finalized dynamic controller

### ISSUE THREAD

**Issue by zroubalik** [knative-sandbox/eventing-autoscaler-keda#2]
If we create a new Event Source, ie. we install Kafka Sources or AWS SQS Source, the parent controller successfully creates a new dynamic controller for this particular Source. But if we unistall this Source, the created dynamic controller is not properly finalized/destroyed. This results in error messages in the log:

```
E0716 16:57:18.095520       1 reflector.go:382] k8s.io/client-go@v12.0.0+incompatible/tools/cache/reflector.go:125: Failed to watch *v1.Source: the server could not find the requested resource
```

This part of the code is probably not properly called:
https://github.com/zroubalik/autoscaler-keda/blob/85186c0c257126f6aa71fac9b26f9b252210e523/pkg/reconciler/crd/reconciler.go#L111-L119

**Comment by aavarghese** [comment]
FYI: I see this error in the logs in BOTH eventing-autoscaler-keda's contoller `controller-xxx -n eventing-autoscaler-keda` AND also in knative-eventing's controller `eventing-controller-xxx  -n knative-eventing` when the CRD is uninstalled.

That expected? 

cc: @zroubalik

**Comment by zroubalik** [comment]
@aavarghese I don't think this is expected. We should talk to some folks familiar with that part of Eventing, maybe @nachocano?

CC @lionelvillard any suggestion?

**Comment by lionelvillard** [comment]
interesting. I've seen this error before in core eventing, can't find the issue. 

@nachocano any ideas?
